marc.py

```diff
@@ -154,11 +155,10 @@ def add_isbn(cls, record, identifier):
             isbn = identifier
         if not isbn:
             _db = Session.object_session(identifier)
-            for identifier_id in identifier.equivalent_identifier_ids()[identifier.id]:
-                equivalent = get_one(_db, Identifier, id=identifier_id)
-                if equivalent.type == Identifier.ISBN:
-                    isbn = equivalent
-                    break
+            identifier_ids = identifier.equivalent_identifier_ids()[identifier.id]
+            isbn = _db.query(Identifier).filter(
+                Identifier.type==Identifier.ISBN).filter(
+                Identifier.id.in_(identifier_ids)).first()
```

Comment:
Maybe add an `order_by(Identifier.id)` so the ISBN is less likely to jump around should the MARC record ever be regenerated.